backend/models/user.js

```diff
@@ -15,6 +15,38 @@ const mongoose = require("mongoose");
 const crypto = require("crypto");
 const uniqueValidator = require("mongoose-unique-validator");
 
+// Define the UserRank schema
+const userRankSchema = new mongoose.Schema({
```

Comment:
You'd better create new file userRank.js for userRankSchema definition.
We will have many schemas and we couldn't put all of them in one file.
It is not a good approach for project maintainability.